File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Entry/CCDAEntry.kt
package com.ccdakit.engine

/**
 * Structured clinical entry inside a C-CDA section.
 *
 * @property id Stable identity for list and tree rendering.
 * @property type XML element type for the entry, such as `observation` or `substanceAdministration`.
 * @property templateIds Template identifiers attached to this entry.
 * @property identifiers Entry identifiers.
 * @property code Primary coded concept for the entry.
 * @property status Entry status code when present.
 * @property effectiveTime Effective time value or low/high fallback parsed from the XML timestamp.
 * @property value Entry value parsed from the direct `value` element.
 * @property textReference Narrative text reference, usually an anchor such as `#problem-1`.
 * @property children Nested entries from `entryRelationship` or `component` children.
 */
data class CCDAEntry(
    val id: String,
    val type: CCDAEntryType,
    val templateIds: List<CCDATemplateId> = emptyList(),
    val identifiers: List<CCDAIdentifier> = emptyList(),
    val code: CCDACodedValue? = null,
    val status: CCDAEntryStatus? = null,
    val effectiveTime: CCDATimestamp? = null,
    val value: CCDAValue? = null,
    val textReference: String? = null,
    val children: List<CCDAEntry> = emptyList(),
)

File: apple/Sources/CCDAEngine/Models/Entry/CCDAEntry.swift
import Foundation

/// Structured clinical entry inside a C-CDA section.
public struct CCDAEntry: Identifiable, Sendable {
    /// Stable identity for list and tree rendering.
    public let id: String
    /// XML element type for the entry, such as observation or substanceAdministration.
    public let type: CCDAEntryType
    /// Template identifiers attached to this entry.
    public let templateIds: [CCDATemplateId]
    /// Entry identifiers.
    public let identifiers: [CCDAIdentifier]
    /// Primary coded concept for the entry.
    public let code: CCDACodedValue?
    /// Entry status code, when present.
    public let status: CCDAEntryStatus?
    /// Effective time value or low/high fallback parsed from the XML timestamp.
    public let effectiveTime: CCDATimestamp?
    /// Entry value parsed from the direct value element.
    public let value: CCDAValue?
    /// Narrative text reference, usually an anchor such as `#problem-1`.
    public let textReference: String?
    /// Nested entries from entryRelationship or component children.
    public let children: [CCDAEntry]

    /// Creates a structured clinical entry.
    public init(
        id: String,
        type: CCDAEntryType,
        templateIds: [CCDATemplateId] = [],
        identifiers: [CCDAIdentifier] = [],
        code: CCDACodedValue? = nil,
        status: CCDAEntryStatus? = nil,
        effectiveTime: CCDATimestamp? = nil,
        value: CCDAValue? = nil,
        textReference: String? = nil,
        children: [CCDAEntry] = []
    ) {
        self.id = id
        self.type = type
        self.templateIds = templateIds
        self.identifiers = identifiers
        self.code = code
        self.status = status
        self.effectiveTime = effectiveTime
        self.value = value
        self.textReference = textReference
        self.children = children
    }
}
